tests/e2e/e2e_local_ratelimit_test.go

```diff
@@ -0,0 +1,177 @@
+		Expect(err).NotTo(HaveOccurred())
+
+		_, err = Td.CreateServiceAccount(destName, &svcAccDef)
+		Expect(err).NotTo(HaveOccurred())
+		_, err = Td.CreatePod(destName, podDef)
+		Expect(err).NotTo(HaveOccurred())
+		dstSvc, err := Td.CreateService(destName, svcDef)
+		Expect(err).NotTo(HaveOccurred())
+
+		// Expect it to be up and running in it's receiver namespace
+		Expect(Td.WaitForPodsRunningReady(destName, 90*time.Second, 1, nil)).To(Succeed())
+
+		srcPod := setupSource(sourceName, false)
+
+		By("Allow traffic via SMI policies")
+		// Deploy allow rule client->server
+		httpRG, trafficTarget := Td.CreateSimpleAllowPolicy(
+			SimpleAllowPolicy{
+				RouteGroupName:    "routes",
+				TrafficTargetName: "test-target",
+
+				SourceNamespace:      sourceName,
+				SourceSVCAccountName: srcPod.Spec.ServiceAccountName,
+
+				DestinationNamespace:      destName,
+				DestinationSvcAccountName: svcAccDef.Name,
+			})
+
+		_, err = Td.CreateHTTPRouteGroup(destName, httpRG)
+		Expect(err).NotTo(HaveOccurred())
+		_, err = Td.CreateTrafficTarget(destName, trafficTarget)
+		Expect(err).NotTo(HaveOccurred())
+
+		clientToServer := HTTPRequestDef{
+			SourceNs:        sourceName,
+			SourcePod:       srcPod.Name,
+			SourceContainer: sourceName,
+
+			Destination: fmt.Sprintf("%s.%s.svc.cluster.local", dstSvc.Name, dstSvc.Namespace),
+		}
+
+		srcToDestStr := fmt.Sprintf("%s -> %s",
+			fmt.Sprintf("%s/%s", sourceName, srcPod.Name),
+			clientToServer.Destination)
+
+		// Send traffic
+		cond := Td.WaitForRepeatedSuccess(func() bool {
+			result := Td.HTTPRequest(clientToServer)
+
+			if result.Err != nil || result.StatusCode != 200 {
+				Td.T.Logf("> (%s) HTTP Req failed with status: %d, err: %s",
+					srcToDestStr, result.StatusCode, result.Err)
+				return false
+			}
+			Td.T.Logf("> (%s) HTTP Req succeeded: %d", srcToDestStr, result.StatusCode)
+			return true
+		}, 5, Td.ReqSuccessTimeout)
+		Expect(cond).To(BeTrue())
+
+		// Rate limit
+		By("Enforce HTTP rate limit at VirtualHost level")
+
+		upstreamTrafficSetting := &policyv1alpha1.UpstreamTrafficSetting{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      dstSvc.Name,
+				Namespace: dstSvc.Namespace,
+			},
+			Spec: policyv1alpha1.UpstreamTrafficSettingSpec{
+				Host: fmt.Sprintf("%s.%s.svc.cluster.local", dstSvc.Name, dstSvc.Namespace),
+				RateLimit: &policyv1alpha1.RateLimitSpec{
+					Local: &policyv1alpha1.LocalRateLimitSpec{
+						HTTP: &policyv1alpha1.HTTPLocalRateLimitSpec{
+							Requests: 1,
+							Unit:     "minute",
+						},
+					},
+				},
+			},
+		}
+
+		upstreamTrafficSetting, err = Td.PolicyClient.PolicyV1alpha1().UpstreamTrafficSettings(upstreamTrafficSetting.Namespace).Create(context.TODO(), upstreamTrafficSetting, metav1.CreateOptions{})
+		Expect(err).ToNot((HaveOccurred()))
+
+		// Expect client to receive 429 (Too Many Requests)
+		cond = Td.WaitForRepeatedSuccess(func() bool {
```

Comment:
Alternatively, we can use the fortio toolkit to set traffic RPS and test rate limiting.